perceval/archive.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2017 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import datetime
+import hashlib
+import json
+import os
+import pickle
+import sqlite3
+
+from grimoirelab.toolkit.datetime import str_to_datetime, datetime_to_utc
+
+from .errors import ArchiveError
+from .utils import DEFAULT_DATETIME
+
+
+STORAGE_EXT = ".sqlite3"
+
+
+def now_to_str():
+    """Return a str value of the current time"""
+
+    now = datetime.datetime.utcnow()
+    return now.strftime("%Y-%m-%d %H:%M:%S")
+
+
+def dict_to_string(json_data):
+    """convert a dict to str"""
+
+    return json.dumps(json_data, sort_keys=True)
+
+
+def make_digest(url, payload, headers, method, stream, verify):
```

Comment:
Not sure if this is the best name for this function. As this is part of `Archive` class and it's only used there I think this should be a static method of that class.

Reply:
ok, what do you think about _make_hash_ ?